realm/src/main/java/io/realm/Realm.java

```diff
@@ -154,6 +155,9 @@
     private static final Map<String, AtomicInteger> globalOpenInstanceCounter =
             new ConcurrentHashMap<String, AtomicInteger>();
 
+    // Thread Pool for all async operations (Query & Write transaction)
+    public static final ExecutorService asyncQueryExecutor = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors() * 2 + 1);
```

Comment:
yep the ThreadPool will use the unbounded queue. do we want to use `ArrayBlockingQueue` to reject subsequent tasks, `RejectedExecutionException` ?